common/OpTestSystem.py

```diff
@@ -1559,37 +1559,6 @@ def sys_active_chip_side(self):
         except OpTestError as e:
             return BMC_CONST.FW_FAILED,BMC_CONST.FW_FAILED
 
-
-
-    ###########################################################################
-    # IPMI Console interfaces
-    ###########################################################################
-
-    ##
-    # @brief It will get the ipmi sol console
-    #
-    # @return l_con @type Object: it is a object of pexpect.spawn class or raise OpTestError
-    #
-    def sys_get_ipmi_console(self):
-        self.l_con = self.bmc.get_host_console()
-        return self.l_con
```

Comment:
I think we still need this function as after BMC reset the console needs to be re-connect. 
Example testcase failure.
```
======================================================================
ERROR [210.719s]: runTest (testcases.BMCResetTorture.Host)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pridhiviraj/new_optests/debmc/op-test-framework/testcases/BMCResetTorture.py", line 95, in runTest
    self.RunBMCReset()
  File "/home/pridhiviraj/new_optests/debmc/op-test-framework/testcases/BMCResetTorture.py", line 69, in RunBMCReset
    con.run_command("uname -a")
  File "/home/pridhiviraj/new_optests/debmc/op-test-framework/common/OpTestIPMI.py", line 312, in run_command
    return self.util.run_command(self, command, timeout, retry)
  File "/home/pridhiviraj/new_optests/debmc/op-test-framework/common/OpTestUtil.py", line 543, in run_command
    raise cf
CommandFailed: Command 'uname -a' exited with -1.
Output:
run_command TIMEOUT or EOF, the command timed out or something, probably a connection issue, retry

----------------------------------------------------------------------
Ran 1 test in 210.719s
```

Reply:
> Example testcase

What machine did you get this on so I can recreate ?

In common/OpTestSystem.py __init__ pulls the console from that:

self.console = self.bmc.get_host_console()

Reply:
You can add below overall fix.
```
diff --git a/common/OpTestIPMI.py b/common/OpTestIPMI.py
index 0f96a29..aa081d5 100644
--- a/common/OpTestIPMI.py
+++ b/common/OpTestIPMI.py
@@ -682,6 +682,7 @@ class OpTestIPMI():
         print ("Applying Cold reset.")
         rc = self.ipmitool.run(BMC_CONST.BMC_COLD_RESET)
         if BMC_CONST.BMC_PASS_COLD_RESET in rc:
+            self.console.state = IPMIConsoleState.DISCONNECTED
             time.sleep(BMC_CONST.SHORT_WAIT_IPL)
             self.util.PingFunc(self.cv_bmcIP, BMC_CONST.PING_RETRY_FOR_STABILITY)
             self.ipmi_wait_for_bmc_runtime()
@@ -744,6 +745,7 @@ class OpTestIPMI():
                             + str(BMC_CONST.BMC_WARM_RESET_DELAY) + "sec")
         rc = self.ipmitool.run(l_cmd)
         if BMC_CONST.BMC_PASS_WARM_RESET in rc:
+            self.console.state = IPMIConsoleState.DISCONNECTED
             print rc
             time.sleep(BMC_CONST.BMC_WARM_RESET_DELAY)
             self.util.PingFunc(self.cv_bmcIP, BMC_CONST.PING_RETRY_FOR_STABILITY)
diff --git a/common/OpTestSystem.py b/common/OpTestSystem.py
index cc35608..4299f07 100644
--- a/common/OpTestSystem.py
+++ b/common/OpTestSystem.py
@@ -1771,10 +1771,12 @@ class OpTestOpenBMCSystem(OpTestSystem):
         self.rest.set_bootdev_to_none()
 
     def sys_warm_reset(self):
+        self.console.state = ConsoleState.DISCONNECTED
         self.rest.bmc_reset()
 
     def sys_cold_reset_bmc(self):
         self.rest.bmc_reset()
+        self.console.state = ConsoleState.DISCONNECTED
 
     def sys_enable_tpm(self):
        self.rest.enable_tpm()
```

Reply:
> overall fix

Will add self.console.close, close will be single place where the proper clearing of the attributes will take place.  This was also done in testcases/BasicIPL.py so I will update that module to remove the .close since we can now do in single place in common/OpTestIPMI.py

Reply:
sure, and also same applicable for common/OpTestSSH based console as well.